products/views.py

```diff
@@ -0,0 +1,76 @@
+from users.models import Position
+from django.views import View
+from django.http  import JsonResponse
+
+from .models      import Category, Product
+
+class MainView(View):
+    def get(self, request):
+        product_info = [{
+            'product_id'    : product.id,
+            'category_name' : product.category.name,
+            'title'         : product.title,
+            'content'       : product.content,
+            'expert_name'   : product.expert.name,
+            'expert_type'   : product.expert.position.name,
+            'expert_image'  : product.expert.image
+        } for product in Product.objects.select_related('category', 'expert', 'expert__position').order_by('-sell_count')]
+
+        OFFSET = 0
+        LIMIT  = 10
+
+        return JsonResponse({'result' : product_info[OFFSET:LIMIT]}, status=200)
+
+class ProductView(View):
+    def get(self , request):
+        category_id = request.GET.get('category')
+        sort        = request.GET.get('sort', '-created_at')
+        category    = Category.objects.get(id=category_id)
+
+        product_info = [{
+            'product_id'    : product.id,
+            'category_name' : product.category.name,
+            'title'         : product.title,
+            'price'         : product.price,
+            'content'       : product.content,
+            'expert_name'   : product.expert.name,
+            'expert_type'   : product.expert.position.name,
+            'expert_image'  : product.expert.image,
+            'hashtag'       : [hashtag.name for hashtag in product.expert.hashtag_set.all()]
+        } for product in category.product_set.select_related('expert', 'category', 'expert__position').prefetch_related('expert__hashtag_set').order_by(sort)]
```

Comment:
이렇게 되면 코드의 가독성이 상대적으로 떨어져보입니다.

```suggestion
    products = select_related() ~~~~~
        } for product in products]
```

와 같이 변경해서 가독성을 높일 수 있습니다!

Reply:
수정 했습니다. 그런데 어떻게 변경하라는건지 잘 모르겠습니다.